orchestration/_tests/test_prune_controller.py

```diff
@@ -0,0 +1,203 @@
+class DummyPruneController(PruneController):
+    """
+    A concrete dummy implementation of PruneController for testing purposes.
+
+    This class uses a local directory (self.base_dir) to simulate file pruning.
+    It provides additional helper methods:
+      - get_files_to_delete(retention): returns files older than the given retention period.
+      - prune_files(retention): deletes files older than the given retention period.
+    """
+    def __init__(self, base_dir: Path):
+        # Create a dummy configuration object with a default beamline_id.
+        dummy_config = type("DummyConfig", (), {})()
+        dummy_config.tc = None
+        dummy_config.beamline_id = "dummy"  # Add a default beamline_id for testing
+        super().__init__(dummy_config)
+        self.base_dir = base_dir
+
+    def get_files_to_delete(self, retention: int):
+        """
+        Return a list of files in self.base_dir whose modification time is older than
+        'retention' days.
+
+        Args:
+            retention (int): Retention period in days. Must be > 0.
+
+        Returns:
+            List[Path]: Files older than the retention period.
+
+        Raises:
+            ValueError: If retention is not positive.
+        """
+        if retention <= 0:
+            raise ValueError("Retention must be a positive number of days")
+        now_ts = datetime.now().timestamp()
+        retention_seconds = retention * 24 * 3600
+        files_to_delete = []
+        for f in self.base_dir.glob("*"):
+            if f.is_file():
+                mod_time = f.stat().st_mtime
+                if now_ts - mod_time > retention_seconds:
+                    files_to_delete.append(f)
+        return files_to_delete
+
+    def prune_files(self, retention: int):
+        """
+        Delete files older than 'retention' days and return the list of deleted files.
+
+        Args:
+            retention (int): Retention period in days. Must be > 0.
+
+        Returns:
+            List[Path]: List of files that were deleted.
+
+        Raises:
+            ValueError: If retention is not positive.
+        """
+        if retention <= 0:
+            raise ValueError("Retention must be a positive number of days")
+        files_to_delete = self.get_files_to_delete(retention)
+        for f in files_to_delete:
+            f.unlink()  # Delete the file
+        return files_to_delete
+
+    def prune(
+        self,
+        file_path: str = None,
+        source_endpoint=None,
+        check_endpoint=None,
+        days_from_now: timedelta = timedelta(0)
+    ) -> bool:
+        """
+        Dummy implementation of the abstract method.
+        (Not used in these tests.)
+        """
+        return True
+
+
+###############################################################################
+# Pytest Fixtures
+###############################################################################
+@pytest.fixture
```

Comment:
pytest already has this fixture: https://docs.pytest.org/en/6.2.x/tmpdir.html#the-tmpdir-fixture,probably use it.